tests/kafkatest/tests/streams/streams_static_membership_test.py

```diff
@@ -0,0 +1,125 @@
+
+        self.producer = VerifiableProducer(self.test_context,
+                                           1,
+                                           self.kafka,
+                                           self.input_topic,
+                                           throughput=1000,
+                                           acks=1)
+
+    def test_rolling_bounces_will_not_trigger_rebalance_under_static_membership(self):
+        self.logger.info("code rolling bounce test started")
+        self.zookeeper.start()
+        self.kafka.start()
+
+        numThreads = 3
+        processor1 = StreamsStaticMembershipService(self.test_context, self.kafka, "consumer-A", numThreads)
+        processor2 = StreamsStaticMembershipService(self.test_context, self.kafka, "consumer-B", numThreads)
+        processor3 = StreamsStaticMembershipService(self.test_context, self.kafka, "consumer-C", numThreads)
+
+        processors = [processor1, processor2, processor3]
+
+        self.producer.start()
+
+        for processor in processors:
+            processor.CLEAN_NODE_ENABLED = False
+            self.set_topics(processor)
+            self.verify_running(processor, 'REBALANCING -> RUNNING')
+
+        self.verify_processing(processors)
+
+        # do several rolling bounces
+        numBounces = 3
+        for i in range(0, numBounces):
+            for processor in processors:
+                self.verify_stopped(processor)
+                self.verify_running(processor, 'REBALANCING -> RUNNING')
+
+        stableGeneration = 3
+        for processor in processors:
+            generations = self.extract_generation_from_logs(processor)
+            for generation in generations[-(numBounces * numThreads):]:
+                assert stableGeneration == int(generation), \
+                    "Stream rolling bounce have caused unexpected generation bump %d" % int(generation)
+
+        self.verify_processing(processors)
+
+        self.stop_processors(processors)
+
+        self.producer.stop()
+        self.kafka.stop()
+        self.zookeeper.stop()
+
+    @staticmethod
+    def extract_generation_from_logs(processor):
+        return list(processor.node.account.ssh_capture("grep \"Successfully joined group with generation\" %s| awk \'{for(i=1;i<=NF;i++) {if ($i == \"generation\") beginning=i+1; if($i== \"(org.apache.kafka.clients.consumer.internals.AbstractCoordinator)\") ending=i }; for (j=beginning;j<ending;j++) printf $j; printf \"\\n\"}\'" % processor.LOG_FILE, allow_fail=True))
+
+    @staticmethod
+    def verify_running(processor, message):
+        node = processor.node
+        with node.account.monitor_log(processor.STDOUT_FILE) as monitor:
+            processor.start()
+            monitor.wait_until(message,
+                               timeout_sec=60,
+                               err_msg="Never saw '%s' message " % message + str(processor.node.account))
+
+    @staticmethod
+    def verify_stopped(processor):
```

Comment:
These two functions seem a bit duplicated with other tests, could we move them to a `util` package under `tests/streams` so that all tests can leverage them?

Reply:
Sounds good!